### PR TITLE
Convert a Text File into Speech

### PR DESCRIPTION
**Pull request by SinghHimadri**
## Issue that this pull request solves
#271 
Closes: # (issue number)
#271 
## Proposed changes

Brief description of what is fixed or changed
The user will be asked if it wants to convert text into speech or a text file. Tkinter is used for GUI the Dialogflow will open and the user can choose the file. After this, the file path is extracted and audio is saved in audioFIles folder and the sent file option opens ans user and send the audio file.
## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface
TERMINAL

Screenshot from 2020-05-27 23-55-49

GUI

Screenshot from 2020-05-27 23-50-49

CHOOSE FILE

Screenshot from 2020-05-27 23-51-02

Screenshot from 2020-05-27 23-51-14

## Other information

Any other information that is important to this pull request

### CODE REVIEW

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- wplay/text_to_speech.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5595346&bid=18329111)

File: wplay/text_to_speech.py [hunk 1]
```diff
@@ -5,34 +5,58 @@
 from wplay.utils.Logger import Logger
 
 from gtts import gTTS
+from tkinter import *
+from tkinter.ttk import * 
+from tkinter.filedialog import askopenfile
```
Comment (rpotter12) [inline wplay/text_to_speech.py:10]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5595346)

File: wplay/text_to_speech.py [hunk 2]
```diff
@@ -5,34 +5,58 @@
 from wplay.utils.Logger import Logger
 
 from gtts import gTTS
+from tkinter import *
+from tkinter.ttk import *
```
Comment (rpotter12) [inline wplay/text_to_speech.py:9]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5595346)

File: wplay/text_to_speech.py [hunk 3]
```diff
@@ -5,34 +5,58 @@
...
+            myobj = gTTS(text=text, lang=language, slow=False)
+            # Saving the converted audio in a mp3 file named
+            myobj.save( audio_file_folder_path / "{}.mp3".format(target))
+
+        else :
+            __logger.info("Converting text file to speech audio file")
+            root = Tk() 
+            root.geometry('200x100')
+            btn = Button(root, text ='Open', command = lambda: open_file()) 
+            btn.pack(side = TOP, pady = 10)
```
Comment (rpotter12) [inline wplay/text_to_speech.py:52]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5595346)

File: wplay/text_to_speech.py [hunk 4]
```diff
@@ -5,34 +5,58 @@
...
+            # Passing the text and language to the engine,
+            myobj = gTTS(text=text, lang=language, slow=False)
+            # Saving the converted audio in a mp3 file named
+            myobj.save( audio_file_folder_path / "{}.mp3".format(target))
+
+        else :
+            __logger.info("Converting text file to speech audio file")
+            root = Tk() 
+            root.geometry('200x100')
+            btn = Button(root, text ='Open', command = lambda: open_file())
```
Comment (rpotter12) [inline wplay/text_to_speech.py:51]: Codacy Issue found: [Lambda may not be necessary](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5595346)

Comment (rpotter12) [inline wplay/text_to_speech.py:51]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5595346)

File: wplay/text_to_speech.py [hunk 5]
```diff
@@ -5,34 +5,58 @@
...
+            text = input("\n\nWrite the text you want to convert to audio file: ")
+
+            # Passing the text and language to the engine,
+            myobj = gTTS(text=text, lang=language, slow=False)
+            # Saving the converted audio in a mp3 file named
+            myobj.save( audio_file_folder_path / "{}.mp3".format(target))
+
+        else :
+            __logger.info("Converting text file to speech audio file")
+            root = Tk()
```
Comment (rpotter12) [inline wplay/text_to_speech.py:49]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5595346)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/text_to_speech.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5595346&bid=18329111)

**Comment by SinghHimadri** [comment]
@rpotter12 Please review my PR.I added a new feature :)

**Comment by rpotter12** [comment]
@SinghHimadri This is not really needed in our software. Converting the text from the command line will be more enough. :)

**Pull request closed without merging**